### ISSUE TITLE
Handle 'pinchcancel' event 

### ISSUE THREAD

**Issue by xoddong** [uber/react-map-gl#856]
We have an application where on zoom out, we immediately hide the markers. Once the zoom out has ended, we fetch new data and redraw the new set of markers on the map. 

While testing this on an iOS device, we noticed something weird.
As we are zooming out with two fingers (pinch), the the map position would suddenly snap into place and no longer zoom out despite my two fingers still being on the screen and in motion.

After this snap, our signal for fetching more data never fires because the last fired interactionState is incorrect. Here is the screenshot of the last fired interactionStateChange. Some of the values are still `true` despite interactionStateChange being "done".
<img width="793" alt="Screen Shot 2019-07-30 at 10 06 37 AM" src="https://user-images.githubusercontent.com/8541335/62149965-bd83d580-b2b1-11e9-9627-c4e50bfaef6c.png">

We narrowed down the issue to this case:

> **When pinch zooming out with at least one finger on a marker, the snapping happens**. 

After some research, we came across issue [#1041](https://github.com/hammerjs/hammer.js/issues/1041) in hammerjs
>**Releasing fingers at the same time triggers `pinchcancel` instead of `pinchend`, which is triggered when releasing one finger, then the other.**

We theorized that when we are zooming out with two fingers with at least one finger on a marker, and then when the marker disappears from zoom out, it ends our pinch event and because we have both fingers still on the screen, `pinchcancel` fires instead of `pinchend`. 

We confirmed this theory in two ways:
1. we stopped hiding the markers on zoom in/out.
2. we added a handler for 'pinchcancel' in `utils/map-controller.js`
```js
var EVENT_TYPES = {
  WHEEL: ['wheel'],
  PAN: ['panstart', 'panmove', 'panend'],
  PINCH: ['pinchstart', 'pinchmove', 'pinchend', 'pinchcancel'],
  DOUBLE_TAP: ['doubletap'],
  KEYBOARD: ['keydown']
};

...

value: function handleEvent(event) {
  this.mapState = this.getMapState();

  switch (event.type) {
    case 'panstart':
      return this._onPanStart(event);

    case 'panmove':
      return this._onPan(event);

    case 'panend':
      return this._onPanEnd(event);

    case 'pinchstart':
      return this._onPinchStart(event);

    case 'pinchmove':
      return this._onPinch(event);

    case 'pinchcancel':
      return this._onPinchEnd(event);

    case 'pinchend':
      return this._onPinchEnd(event);

    case 'doubletap':
      return this._onDoubleTap(event);

    case 'wheel':
      return this._onWheel(event);

    case 'keydown':
      return this._onKeyDown(event);

    default:
      return false;
  }
}
```

Both fixed this issue.
We have currently implemented solution 1 on our end. However, it would be nice if the library handled `pinchcancel` event!

**Comment by Pessimistress** [comment]
Thank you @xoddong! Would you open a PR to introduce `pinchcancel` handling?

**Comment by xoddong** [comment]
Will do!

**Comment by Pessimistress** [comment]
Fixed in 4.1.11/5.0.8
